src/snowflake/snowpark/modin/pandas/dataframe.py

```diff
@@ -227,9 +227,21 @@ def __init__(
                     if dtype is not None:
                         new_qc = new_qc.astype({col: dtype for col in new_qc.columns})
                     if index is not None:
-                        new_qc = new_qc.reindex(axis=0, labels=index)
+                        # TODO: SNOW-1372242: Remove instances of to_pandas when lazy index is implemented
+                        new_qc = new_qc.reindex(
```

Comment:
@sfc-gh-dpetersohn is making changes like this in `src/snowflake/snowpark/modin/pandas` ok?